edx_event_bus_kafka/internal/consumer.py

```diff
@@ -278,8 +304,10 @@ def _consume_indefinitely(self):
                     msg = self.consumer.poll(timeout=CONSUMER_POLL_TIMEOUT)
                     if msg is not None:
                         with function_trace('_consume_indefinitely_consume_single_message'):
-                            # Before processing, make sure our db connection is still active
-                            _reconnect_to_db_if_needed()
+                            # Before processing, make sure our application state is similar to
+                            # that of a new Django request. (Mimic setup/teardown.)
```

Comment:
Non-blocking question: By "setup/teardown", are you referring to something related to Django requests, or just unit tests or some such? If Django requests, I have https://github.com/openedx/openedx-events/issues/236 open, because I'm wondering if there isn't a more general solution to this problem.

Reply:
1. It wouldn't hurt to link to that issue and add some clarity.
2. It actually feels like `prepare_for_new_work_cycle` probably belongs in openedx-events, which would give this benefit to Redis for free. Maybe we could create an issue for that, and let the Redis team know about it. In place of copying this change, maybe they can implement a more shareable solution?

Reply:
Created https://github.com/openedx/openedx-events/issues/270